### PR TITLE
Alter Warp Sync market wording and remove 777 :[

### PR DESCRIPTION
**Pull request by nuevoalex**
Addresses:

#5082
#5123
#5124

### CODE REVIEW

**Review by epheph — APPROVED** [review]
👍  but #5046 needs to be merged first

**Comment by MicahZoltu** [comment]
I feel like you are trolling me asking me to review this...

**Comment by nuevoalex** [comment]
❤️ u @MicahZoltu

File: packages/augur-core/source/contracts/Cash.sol [hunk 1]
```diff
@@ -137,7 +132,7 @@ contract Cash is ITyped, ICash, ICashFaucet {
 
     function mint(address _target, uint256 _amount) internal returns (bool) {
         balances[_target] = balances[_target].add(_amount);
-        supply = supply.add(_amount);
+        totalSupply = totalSupply.add(_amount);
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/Cash.sol:135]: Where is this `totalSupply` variable defined?  This contract appears to only derive from some interfaces (not contracts) and you removed the `totalSupply` function and didn't replace it with a `totalSupply` member variable...

File: packages/augur-core/source/contracts/legacy_reputation/OldLegacyRepToken.sol [hunk 2]
```diff
@@ -107,8 +107,8 @@ contract OldLegacyReputationToken is DelegationTarget, ITyped, Initializable, Va
     }
 
     function transferFrom(address _from, address _to, uint _value) public whenNotMigratingFromLegacy returns (bool) {
-        require(_value <= _allowances[_from][msg.sender], "Not enough funds allowed");
-        _allowances[_from][msg.sender] = _allowances[_from][msg.sender].sub(_value);
+        require(_value <= allowances[_from][msg.sender], "Not enough funds allowed");
+        allowances[_from][msg.sender] = allowances[_from][msg.sender].sub(_value);
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/legacy_reputation/OldLegacyRepToken.sol:111]: No changes should be made to `OldLegacyRepToken` or `LegacyRepToken` as they are meant to represent contracts live on mainnet.  Even though this particular change may just be internal details, I think it is generally a bad practice to be mutating those two contracts.

File: packages/augur-core/source/contracts/libraries/token/ERC20.sol [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+ * Additionally, an {Approval} event is emitted on calls to {transferFrom}.
+ * This allows applications to reconstruct the allowance for all accounts just
+ * by listening to said events. Other implementations of the EIP may not emit
+ * these events, as it isn't required by the specification.
+ *
+ * Finally, the non-standard {decreaseAllowance} and {increaseAllowance}
+ * functions have been added to mitigate the well-known issues around setting
+ * allowances. See {IERC20-approve}.
+ */
+contract ERC20 is IERC20 {
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/libraries/token/ERC20.sol:31]: Was this pulled from OpenZeppelin or somewhere else?  The prolific comments suggests that it wasn't hand-rolled in-house.  😉

Comment (MicahZoltu) [reply packages/augur-core/source/contracts/libraries/token/ERC20.sol:31]: Ah, looking at the comments it appears OZ.  Might be worth including a permalink to the OZ contract.

File: packages/augur-core/source/contracts/Cash.sol (hunk 1 above)
Comment (nuevoalex) [reply packages/augur-core/source/contracts/Cash.sol:135]: Our interfaces are actually contracts. Its defined in IERC20.sol

File: packages/augur-core/source/contracts/libraries/token/IERC20.sol [hunk 4]
```diff
@@ -2,7 +2,8 @@ pragma solidity 0.5.10;
 
 
 contract IERC20 {
-    function totalSupply() public view returns (uint256);
+    uint256 public totalSupply;
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/libraries/token/IERC20.sol:5]: I believe this change is incorrect.  Interfaces should not have members on them (even when they are technically contracts), they should only have functions.  In Solidity it is legal to do the following:
```
contract IERC20 {
    function totalSupply() external view returns (uint256);
}
contract ERC20 is IERC20 {
    uint256 public totalSupply;
}
```
Note: For reasons not entirely clear to me, the interface method needs to be `external`, not `public` for the above to work.

File: packages/augur-core/source/contracts/reporting/OICash.sol [hunk 5]
```diff
@@ -21,16 +21,16 @@ contract OICash is VariableSupplyToken, Initializable, IOICash {
...
         augur = _augur;
         cash = ICash(_augur.lookup("Cash"));
         cash.approve(address(_augur), MAX_APPROVAL_AMOUNT);
         shareToken = IShareToken(_augur.lookup("ShareToken"));
         require(shareToken != IShareToken(0));
         universe = _universe;
-        erc1820Registry = IERC1820Registry(_erc1820RegistryAddress);
-        initialize1820InterfaceImplementations();
+
+        initializeMKRShutdownHandler(_augur.lookup("DaiVat"), _augur.lookup("Cash"));
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/reporting/OICash.sol:33]: Isn't this line related to an unmerged branch?  Perhaps a mistake when fiddling with branch changes?

Comment (MicahZoltu) [reply packages/augur-core/source/contracts/reporting/OICash.sol:33]: Note: Same question for several other files.

**Review by MicahZoltu — COMMENTED** [review]
I am not a dignified enough person to be willing to approve this PR due to my strong disagreement with the removal of ERC777.  I am, however, a dignified enough person to not reject the PR.

I do recommend fixing the `interface` problem I mentioned, and I think this PR has some changes from another branch mixed in that need to be cleaned up before merge.

File: packages/augur-core/source/contracts/libraries/token/IERC20.sol (hunk 4 above)
Comment (nuevoalex) [reply packages/augur-core/source/contracts/libraries/token/IERC20.sol:5]: It is not legal to do that in Solidity.

**Comment by epheph** [comment]
Unlike @MicahZoltu , my approval comes cheap. 🤠

**Pull request merged**